resources/type-system/flow-analysis.md

```diff
@@ -220,13 +226,36 @@ regardless of their static types.
 
 We also make use of the following auxiliary functions:
 
-- `joinV(VM1, VM2)`, where `VM1` and `VM2` are variable models, represents the
-  union of two variable models, defined as follows:
+- `joinDeclaredType(d1, d2)` is defined as:
+  - if `d1` and `d2` are declared types (the variable was declared with a type,
+    or has a type inferred from its initializer), then `d3`.
+  - Otherwise, `d1` and `d2` are **initialization inferred types** and are
+    merged as follows:
+    - If `d1` and `d2` are both missing, then `d3` is missing.
+    - Otherwise if `r1` and `r2` are both true and either `d1` or `d2` is
+      missing, and the variable in question was not declared `late` then it is
+      an error.
+    - Otherwise if `d1` and `d2` are equal then `d3` is `d1`, and is final if
```

Comment:
Confusing, maybe 'is `d1` (which may or may not be final)'. I assume that `equal` implies they are both final, or none of them is final.